packages/pages/src/components/analytics/analytics.test.tsx

```diff
@@ -0,0 +1,150 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { TemplateProps } from "../../common/src/template/types";
+import { AnalyticsProvider } from "./provider";
+import { useAnalytics } from "./hooks";
+import { AnalyticsScope } from "./scope";
+
+global.fetch = jest.fn().mockImplementation(
+  jest.fn(() => {
+    return Promise.resolve({ status: 200 });
+  }) as jest.Mock
+);
+
+jest.spyOn(global, "fetch");
+
+const baseProps: TemplateProps = {
+  document: {
+    __: {
+      entityPageSet: {},
+      name: "foo",
+    },
+    uid: 0,
+    businessId: 0,
+    siteId: 0,
+  },
+  __meta: { mode: "development" },
+};
+
+const currentProcess = global.process;
+
+beforeAll(() => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  global.process = undefined;
+});
+
+beforeEach(() => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  global.fetch.mockClear();
+});
+
+afterAll(() => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  delete global.fetch;
+  global.process = currentProcess;
+});
+
+describe("Analytics", () => {
+  it("should fire a page view", () => {
+    render(<AnalyticsProvider templateData={baseProps} requireOptIn={false} />);
+    expect(global.fetch).toHaveBeenCalledTimes(1);
+  });
+  it("should not fire a page view when opt in is required", () => {
```

Comment:
Can you add empty lines between the tests?

Reply:
Sure! can you add to eslint?